stm32f0_hal/src/qei.rs

```diff
@@ -0,0 +1,483 @@
+use cortex_m;
+use stm32f0x2::{TIM16 as TIMER16, TIM17 as TIMER17, TIM2 as TIMER2, TIM3 as TIMER3,
+                TIM6 as TIMER6, USART3 as UART3, GPIOA, GPIOB, GPIOC, NVIC, RCC};
+use stm32f0x2::interrupt::*;
+use core::ptr;
+
+use core;
+
+const FREQUENCY: u32 = 48000000;
+const STEP1: u16 = 2249;
+const STEP2: u16 = 2249;
+const DEGREEBYSTEP1: f32 = (360 as f32 / STEP1 as f32);
+const DEGREEBYSTEP2: f32 = (360 as f32 / STEP2 as f32);
+
+static mut DEGREE1: f32 = 0.0;
+static mut PREVIOUS_DEGREE1: f32 = 0.0;
+static mut DELTA_DEGREE1: f32 = 0.0;
+
+static mut DEGREE2: f32 = 0.0;
+static mut PREVIOUS_DEGREE2: f32 = 0.0;
+static mut DELTA_DEGREE2: f32 = 0.0;
+
+pub fn setup_pwm1(frequency_khz: u16) {
```

Comment:
@csagonero you should review these function with attention, I do it without knowing anything about these timers...